cee_syslog_handler/__init__.py

```diff
@@ -271,3 +271,50 @@ def filter(self, record):
         """
         found = self._pattern.search(record.getMessage())
         return not found
+
+
+class RegexRedactFilter(logging.Filter):
+    """
+    This filter redacts parts of log messages and exception traces
+    by substituting through a provided regular expression.
+    Use with caution: with great power comes great responsibility
+    """
+    def __init__(self, filter_regex=None,
+                 replace_string="<redacted>"):
+        super(RegexRedactFilter, self).__init__()
+        self._pattern = re.compile(filter_regex)
+        self._replacement = replace_string
+
+    def redact(self, string):
+        return re.sub(self._pattern, self._replacement, string)
+
+    def filter(self, record):
+        if not self._pattern.search(record.getMessage())\
+                and not record.exc_info:
```

Comment:
It seems weird that we check for `exc_info` here but also then have guards on `exc_text` and `stack_info` below.